pkg/types/baremetal/defaults/platform.go

```diff
@@ -142,4 +142,16 @@ func SetPlatformDefaults(p *baremetal.Platform, c *types.InstallConfig) {
 			return p.Hosts[i].CompareByRole(p.Hosts[j])
 		})
 	}
+
+	var numWorkers int64
+
+	for _, h := range p.Hosts {
+		if h.IsWorker() {
```

Comment:
IIRC specifying a role for each host is not mandatory, so a more robust way is to count the total number of hosts and subtract the number of control plane replicas.

Reply:
i.e. `len(p.Hosts) - *c.ControlPlane.Replicas`